marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -82,49 +90,229 @@ renderCurrentState walletDetails state =
                       [ text "1hr 2mins left" ]
                   ]
               ]
-          , div [ classNames [ "py-2" ] ]
-              [ renderTasks walletDetails state
-              ]
+          , if currentTab == Tasks then
+              renderTasks state
+            else
+              renderBalances state
           ]
       ]
 
-renderTasks :: forall p. WalletDetails -> State -> HTML p Action
-renderTasks walletDetails state =
+-- This helper function expands actions that can be taken by anybody,
+-- then groups by participant and sorts it so that the owner starts first and the rest go
+-- in alphabetical order
+expandAndGroupByRole ::
+  Maybe Party ->
+  Set Party ->
+  Array NamedAction ->
+  Array (Tuple Party (Array NamedAction))
+expandAndGroupByRole mActiveUserParty allParticipants actions =
+  expandedActions
+    # Array.sortBy currentPartyFirst
+    # Array.groupBy sameParty
+    # map extractGroupedParty
+  where
+  -- If an action has a participant, just use that, if it doesn't expand it to all
+  -- participants
+  expandedActions :: Array (Tuple Party NamedAction)
+  expandedActions =
+    actions
+      # foldMap \action -> case getActionParticipant action of
+          Just participant -> [ participant /\ action ]
+          Nothing -> Set.toUnfoldable allParticipants <#> \participant -> participant /\ action
+
+  currentPartyFirst = \(Tuple party1 _) (Tuple party2 _) ->
+    if (Just party1) == mActiveUserParty then
+      LT
+    else
+      if (Just party2) == mActiveUserParty then
+        GT
+      else
+        compare party1 party2
```

Comment:
I think this can also be written like:
```
  currentPartyFirst (Tuple party1 _) (Tuple party2 _)
    | Just party1 == mActiveUserParty = LT
    | Just party2 == mActiveUserParty = GT
    | otherwise = compare party1 party2
```

Reply:
nice one 👍🏻 

Just changed, will enter in the next PR